metricbeat/module/golang/expvar/expvar_integration_test.go

```diff
@@ -0,0 +1,80 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build integration
+
+package expvar
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/beats/libbeat/tests/compose"
+	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
+)
+
+func TestData(t *testing.T) {
+	compose.EnsureUp(t, "golang")
```

Comment:
Would it work to start an http endpoint with pprof here in the test itself?

Reply:
you mean serving an example json doc? Yes. And I think we should this in most http based modules moving forward (see last point here: https://github.com/elastic/beats/issues/10238). But for now this is the same as for all the other modules and mainly meant to have something that works :-)

Reply:
I mean that instead of using docker, we may serve the profile of the test itself with `import _ "net/http/pprof"` and `httptest.NewServer(nil)`.

Reply:
Well, but this way we could test only one version of go, lets leave it with docker, and maybe in the future serving example json docs :+1: